esphome/components/ota/__init__.py

```diff
@@ -41,6 +42,7 @@
     {
         cv.GenerateID(): cv.declare_id(OTAComponent),
         cv.Optional(CONF_SAFE_MODE, default=True): cv.boolean,
+        cv.Optional(CONF_VERSION, default=1): cv.one_of(1, 2, int=True),
```

Comment:
Lets just default to the newest, otherwise nobody will use it =)
```suggestion
        cv.Optional(CONF_VERSION, default=2): cv.one_of(1, 2, int=True),
```